.github/workflows/acquia-backup-databases.yml

```diff
@@ -0,0 +1,99 @@
+name: Workflow to backup databases on Acquia.
+
+on:
+  workflow_call:
+    inputs:
+      filter:
+        description: 'Optionally, provide a filter for the databases. See: https://docs.acquia.com/acquia-cli/commands/#api-environments-database-list'
+        type: string
+        required: false
+      php_version:
+        description: 'Optionally, specify a PHP version to install. Default: 8.2.'
+        type: string
+        required: false
+        default: '8.2'
+    secrets:
+      ENVIRONMENT_ID:
+        description: 'Acquia environment ID.'
+        required: true
+      ACLI_CLIENT_ID:
+        description: 'The client ID for authenticating with ACLI.'
+        required: true
+      ACLI_CLIENT_SECRET:
+        description: 'The client secret for authenticating with ACLI.'
+        required: true
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  backup-databases:
+    name: 'Backup Acquia databases'
+    runs-on: ubuntu-latest
+    outputs:
+      databases: ${{ steps.get_databases.outputs.database_list }}
+      db_list: ${{ steps.get_databases.outputs.db_list }}
+    steps:
+      - name: Checking out code
+        uses: actions/checkout@v3
+
+      - name: Installing PHP
+        uses: shivammathur/setup-php@master
+        with:
+          php-version: ${{ inputs.php_version }}
+
+      - name: Installing dependencies
+        run: |
+          curl -OL https://github.com/acquia/cli/releases/download/2.13.0/acli.phar
+          echo "ACLI downloaded!"
+
+      - name: Authorizing with ACLI
+        run: |
+          php acli.phar auth:login -n \
+            --key="${{ secrets.ACLI_CLIENT_ID }}" \
+            --secret="${{ secrets.ACLI_CLIENT_SECRET }}"
+
+      - name: Retrieving all database names for this environment
+        id: get_databases
+        # Gets databases filtering by a name. If no filter is provided, it'll
+        # retrieve all databases.
+        run: |
+          # The first time we run api:environments:database-list acli very
+          # kindly also outputs a string asking the user to share
+          # anonymous user data along with the JSON output causing an error
+          # further down the chain - this simply mitigates that.
+          acsf_request=$(php acli.phar api:environments:database-list \
+            --filter='${{ inputs.filter }}' \
+            --limit='1' \
+            ${{ secrets.ENVIRONMENT_ID }})
+
+          # Gets a list of databases and uses jq to filter the output
+          # to the name value only.
+          database_list=$(php acli.phar api:environments:database-list \
+            --filter='${{ inputs.filter }}' \
+            ${{ secrets.ENVIRONMENT_ID }} \
+            | jq '.[] | .name')
+
+          # Processes the output of databases to allow the data to
+          # be passed to the 'Backing up databases' step.
+          delimiter="$(openssl rand -hex 8)"
+          echo "db_list<<${delimiter}" >> "${GITHUB_OUTPUT}"
```

Comment:
Oh, you could/should change this to `database_list` and remove the `db_list` value. (Or if you want to keep `database_list` as a temporary variable, then remove it from the outputs at the start of the job.) No point in having two variables in the output if you're only using one.